### ISSUE TITLE
'Download as CBZ' option 'broken' for large files

### ISSUE THREAD

**Issue by wisewtf** [tachiyomiorg/tachiyomi#6874]
### Steps to reproduce

Hello,

I've just tested with Discord support that this is the problem.

I'm trying to download a relatively large comic book from my Mango server from TachiJ2K with the Mango extension. The file is a .cbz file of 350MB with 352 pages.

What happens is that during download, when the 'Download as CBZ' option is enabled, it gets stuck at the last page for maybe 30 seconds and then it restarts the download in a loop, and never manages to actually download the book.

The same thing **sorta** happens with something half as big. I tried with a manga (also .cbz) with 150 pages, this time tho, it also got stuck at the last page, but it hanged there whilst the manga was actually counted as downloaded in my Tachiyomi catalog. And I had to cancel all downloads manually, but the book was still downloaded.

None of this happens when I download the first book with the 'Download as CBZ' option disabled. It downloads fine entirely.

This should be easily reproducible, but let me know if you need any more info.

### Expected behavior

I expect the large book to be downloaded

### Actual behavior

It gets stuck on the last page when the 'Download as CBZ' option is enabled in settings and then restarts the download on a loop

### Tachiyomi version

J2K V 1.4.4

### Android version

Android 10

### Device

Galaxy Tab 10

### Other details

Let me know if I should open this issue in the J2K repo instead. I'm assuming the CBZ code is the same.

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open issue.
- [X] I have written a short but informative title.
- [X] If this is an issue with an extension, I should be opening an issue in the [extensions repository](https://github.com/tachiyomiorg/tachiyomi-extensions/issues/new/choose).
- [X] I have tried the [troubleshooting guide](https://tachiyomi.org/help/guides/troubleshooting/).
- [X] I have updated the app to version **[0.13.1](https://github.com/tachiyomiorg/tachiyomi/releases/latest)**.
- [X] I have updated all installed extensions.
- [X] I will fill out all of the requested information in this form.

**Comment by alessandrojean** [comment]
Managed to reproduce this a while ago when downloading a larger set from E-hentai extension too. The downloader seems to fail when there's a lot of pages in the chapter somehow.

**Comment by Seishirou101** [comment]
can y'all provide some examples to test? I attempted with two different downloads ranging from 350~ to 600~ pages but they downloaded fine, so it may have to deal with the size of the download instead of the page count. The files were only 50 to 60 mb.

**Comment by alessandrojean** [comment]
The image set I tried had around 900 pages and the size was almost 2 GB (I think). It was a while ago, and I don't remember which one it was, so I think any image set with a size around that should trigger the issue.

**Comment by Seishirou101** [comment]
I can't actually reproduce this, I just tried with a file that was 1.3gb with 396 pages on both j2k 1.4.4 and latest preview r4224.  How long did it take for it to restart and start downloading again? Are you using a sd card? Can you provide a crashlog to see if it has anything useful?

**Comment by wisewtf** [comment]
Hello,

I honestly have no idea. I have failed EVERY SINGLE TEST from a while ago and yesterday. I try not to gather information, just with reactivating the CBZ option, and it downloaded fine. It took a while at the last page, but it didn't restart.

At this point I don't know what to say.

- I am using an SD Card (SanDisk SDHC 16GB).
- As I've stated in my issue, it takes about 30 seconds for it to restart downloading. Sometimes less.
- I can't replicate it anymore, so the crash logs would be useless. I've checked them and there's only info about my Komga extension giving errors because my server is down.
